### PR TITLE
Pil 429 getorelse

### PR DESCRIPTION
**Pull request by olakunlekuye2**

### CODE REVIEW

File: app/uk/gov/hmrc/pillar2/models/hods/subscription/common/SubscriptionResponse.scala [hunk 1]
```diff
@@ -70,3 +70,12 @@ case class SecondaryContactDetails(
 object SecondaryContactDetails {
   implicit val format: OFormat[SecondaryContactDetails] = Json.format[SecondaryContactDetails]
 }
+
+case class DashboardInfo(
+  organisationName: String,
+  registrationDate: LocalDate
```
Comment (AKKUM) [inline app/uk/gov/hmrc/pillar2/models/hods/subscription/common/SubscriptionResponse.scala:76]: Not sure LocalDate will work? From ETMP data comes with time. Please check the specs

File: app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala [hunk 2]
```diff
@@ -423,26 +423,46 @@ class SubscriptionService @Inject() (
...
-      addressLine4 = sub.upeCorrespAddressDetails.addressLine4,
-      postalCode = sub.upeCorrespAddressDetails.postCode.getOrElse(""),
+      addressLine2 = sub.upeCorrespAddressDetails.addressLine2.filter(_.nonEmpty).orElse(Some("N/A")),
+      addressLine3 = sub.upeCorrespAddressDetails.addressLine3
+        .collect {
+          case str if str.nonEmpty => str
+        }
+        .getOrElse("N/A"),
+      addressLine4 = sub.upeCorrespAddressDetails.addressLine4.filter(_.nonEmpty).orElse(Some("N/A")),
+      postalCode = sub.upeCorrespAddressDetails.postCode.filter(_.nonEmpty).orElse(Some("N/A")),
```
Comment (AKKUM) [inline app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala:451]: Did we not have only AddressLine3 optional problem

Comment (olakunlekuye2) [reply app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala:451]: No the date for NonUKAddress which has optional fileds  addressLine2, addressLine4 and postalCode gets its data from UpeCorrespAddressDetails which has addressLine2 - addressLine4 and postCode as optional

Comment (olakunlekuye2) [reply app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala:451]: Got rid of the getOrElse

File: app/uk/gov/hmrc/pillar2/models/hods/subscription/common/SubscriptionResponse.scala (hunk 1 above)
Comment (olakunlekuye2) [reply app/uk/gov/hmrc/pillar2/models/hods/subscription/common/SubscriptionResponse.scala:76]: There is nothing in the spec or epid that is specific except for this :                 
registrationDate: 
                  allOf:
                    - $ref: '#/components/schemas/dateType'
                    - description: Registration date

File: app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala [hunk 3]
```diff
@@ -442,11 +442,10 @@ class SubscriptionService @Inject() (
...
-      addressLine3 = sub.upeCorrespAddressDetails.addressLine3
-        .collect {
-          case str if str.nonEmpty => str
-        }
-        .getOrElse("N/A"),
+      addressLine3 = sub.upeCorrespAddressDetails.addressLine3 match {
+        case Some(str) if str.nonEmpty => str
+        case _                         => "N/A"
+      },
       addressLine4 = sub.upeCorrespAddressDetails.addressLine4.filter(_.nonEmpty).orElse(Some("N/A")),
```
[the remark this answers is not in the document]
Comment (olakunlekuye2) [reply app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala:449]: No I'll get rid of the getorelse

**Review by AKKUM — APPROVED** [review]

**Pull request merged**